enrichments/trace/internal/elastic/span.go

```diff
@@ -237,6 +238,39 @@ func (s *spanEnrichmentContext) enrichTransaction(
 	}
 }
 
+// In OTel a root span can represent an outgoing call or a producer span
+// In such cases, the span is still mapped into a transaction, but such spans are enriched
+// with additional attributes that are specific to the outgoing call or producer span.
+func (s *spanEnrichmentContext) enrichExitSpanTransaction(
+	span ptrace.Span,
+	cfg config.Config,
+) {
+	if span.Kind() == ptrace.SpanKindClient || span.Kind() == ptrace.SpanKindProducer {
```

Comment:
I'm not sure we can rely on that condition, tbh.

Since that entire situation we try to handle here is an instrumentation bug in itself (IN THEORY: there never should be a single span that is entry and exit at the same time, but IN PRACTICE that happens in some situations), I don't think we should rely on `SpanKind` at all, tbh.

For example, what if the SpanKind == `SpanKindServer`, but the span still represents entry and exit at the same time?

How about, we use heuristics similar to how we do it for deriving `span.type` to identify whether a span is an exit span or not. For example, checking for certain attributes that would only make sense on exit spans but not on entry spans.

Reply:
This is a valid point and was thinking about this as well. My idea was to be on the safe side and only enrich real exit spans with the necessary attributes.

I understand we have this issue with nginx proxy spans that are a single span for both incoming and outgoing calls. I don't know what's the span type there, but if it's `SpanKindServer` then we would not enrich that indeed.

But that use-case is already a bug and I'm not sure how much we want to unwind such bugs.

> For example, what if the SpanKind == SpanKindServer, but the span still represents entry and exit at the same time?

I think that's bug in the instrumentation. [SemConv says ](https://opentelemetry.io/docs/specs/semconv/http/http-spans/):
> For an HTTP client span, SpanKind MUST be Client.

and 

> For an HTTP server span, SpanKind MUST be Server.

So for HTTP that goes against the spec and I'm not sure how much we should fight these bugs.

> How about, we use heuristics similar to how we do it for deriving span.type to identify whether a span is an exit span or not. For example, checking for certain attributes that would only make sense on exit spans but not on entry spans.

We could explore that, but I'm not sure how reliable it is. The check if something is an exit span is a bit different - there is no differentiation in that check between incoming and outgoing spans, because incomings are already assumed to be transactions, so all those cases are already filtered out at that point by only doing the check for spans.

I just quickly looked at HTTP spans, what I see, the server side only has `url.path` which is only required in server spans, and is not present in client spans. And then `url.full` is only on theclient-side. But if spankind is already invalid, I'm not sure how much we can rely on attributes. E.g. what if both are present?

Overall I feel we'd end up with a messy heuristic which may not work in all cases and may not even unwind these bugs in a way we'd want to.

If we relax this check here, I think the risk we run here is that we may categorize some incoming calls as exit spans and we start calculating dependencies for those. That'd be very bad. If we can come up with something to avoid that, then I'd feel more confident to relax this.